### DISCUSSION TITLE
Remote Server, Session, and User Management

### DISCUSSION THREAD

**Discussion by Rob-NY** [jamulussoftware/jamulus#1990] [General]
Back in June I posted #1847 and shared some code for additional messages in Jamulus core to get basic management information needed for server operators (users w/ their IP, Recording state, etc).  The approach is in line with how Jamulus processes messages.  (https://github.com/Rob-NY/jamulus/tree/server-status).

I've recently been testing application code that makes use of these extensions and I wanted to introduce it here so others could see what I had in mind.

vJammers.com (https://vjammers.com) will let you take control of sessions, manage recordings, manage users in the room, restart the server, etc.   You can try it out anonymously by logging in with demo@vjammers.com and password demo.  Sessions are limited to 30 minutes and, since I wouldn't know who you are, session recordings are not available.  Since credentials are shared, play at your own risk using "demo". 

Anyone interested in *really* trying it out - 4 hour sessions, recording, downloading recordings, private rooms, booting bad actors, etc., I'm happy to grant access -- just email beta@vjammers.com as outlined on the site.  

This is a free service at this time -- giving back to all the folks around the world for making the last 17 months of Covid bearable with Jamulus.

I have servers running in NY/USA, Atlanta/USA, Los Angeles/USA,  London/UK, Frankfurt/Germany.  All are registered on Any Genre 2.  If you'd like a different region spun up, I can try to accommodate based upon VULTR centers.

I'm interested in feedback on the app as well as the proposed Jamulus code.  This is also tied to #1962 and #1958.

**Comment by ann0see** [comment]
Hi Rob!

Thanks for showing a working implementation! The project would benefit a lot from having some more control of the server and client. We're having MQTT, JSON-RPC and your approach [but a little bit differently than the other proposals] (and they all seem to be working! Could you please comment on https://github.com/jamulussoftware/jamulus/pull/1975 and look at different implementations?)

Edit: you might even want to open a draft PR with your code and link the other related PRs

**Comment by ann0see** [comment]
@Rob-NY just a quick question here: do you think you can adapt your scripts to use JSON-RPC (and publish at least parts of your idea/code)? I think many would benefit from a Web-Management console. 
Since JSON-RPC seems to be more feature-rich and "standard" I assume we‘re going with this approach. However I wanted to say a big thank you to you – also since you already have working software.

**Reply by ann0see** [reply]
Concerning the feedback: I‘d really like having that app on our server (but preferably without having to install WordPress)

**Reply by Rob-NY** [reply]
@ann0see  - I'm not sure my scripts would make much sense since they're baked into a much larger platform.  But, if I understand your ask, are you looking for some basic web-based management console that can be offered up to users that demonstrate the RPC capabilities while still providing some stand alone value?  This isn't a bad idea and I'd be willing to try and put something together.

**Reply by ann0see** [reply]
> are you looking for some basic web-based management console that can be offered up to users that demonstrate the RPC capabilities while still providing some stand alone value?

Exactly. Unfortunately I won’t have much time in the next few months to spend on a coding project, but it shouldn’t be too much work

**Reply by Rob-NY** [reply]
I'll take a look.  It is worth noting that the current constraint with the RPC solution to remove the ability to specify an IP address via command line argument and ONLY listen on the loopback means that a proxy will have to live on the same machine (to keep it simple for the average user and not get into tunneling, etc..).  This also means that you can't execute the RPC directly via client-side JS (which would be much cleaner).  That restriction should be revisited.

**Reply by ann0see** [reply]
Yes, that’s a security decision – however feel free to hint at that issue again in the PR.

**Reply by ann0see** [reply]
I have some PHP authentification stuff laying around (my first steps into programming) with user management, DB access,… if you need anything I can publish the files somewhere, although they’re probably far from "quality code".

**Comment by Rob-NY** [comment]
I got it covered, but thanks.
